src/repositories/repository.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  BaseEntity,
+  EntityManager,
+  EntityTarget,
+  FindOptionsRelations,
+  FindOptionsWhere,
+  Repository,
+} from 'typeorm';
+import { QueryDeepPartialEntity } from 'typeorm/query-builder/QueryPartialEntity.js';
+
+import { AppDataSource } from '../plugins/datasource';
+import { IdPkBaseEntity } from '../services/item/plugins/app/appAction/appAction';
+import {
+  EntryNotFoundAfterInsertException,
+  IllegalArgumentException,
+  InsertionException,
+} from './errors';
+
+export abstract class AbstractRepository<T extends BaseEntity> {
```

Comment:
Should all those repositories have their own file? Also I was wondering if we should go for the file naming convention of using a capital letter? ie. `../../ReadRepository.ts`?

Reply:
I think it's a good idea to have one repo per file! concerning the convention, has always I think what matters is that the whole team accepts and follows the convention. Personally, I like the `PascalCase` convention, but I'm open to any other file convention like the `kebab-case` or `camelCase`.

Reply:
Personnally, when the file is intented to be a class file, I'd prefer PascalCase. However, I'm okay with any other convention !